docs/source/optim.rst

```diff
@@ -129,6 +129,50 @@ Algorithms
     Rprop
     SGD
 
+Many of our algorithms have various implementations optimized for performance, so
+we attempt to default to the faster foreach implementation if no particular
```

Comment:
nit: you didn't really define what "foreach" mean at this point. This can just say we default to the generally fastest implementation for the current device.